nexus/db-queries/src/db/datastore/db_metadata.rs

```diff
@@ -8,21 +8,206 @@ use super::DataStore;
 use crate::db;
 use crate::db::error::public_error_from_diesel_pool;
 use crate::db::error::ErrorHandler;
-use async_bb8_diesel::AsyncRunQueryDsl;
+use async_bb8_diesel::{AsyncRunQueryDsl, AsyncSimpleConnection};
+use chrono::Utc;
 use diesel::prelude::*;
 use omicron_common::api::external::Error;
 use omicron_common::api::external::SemverVersion;
+use omicron_common::nexus_config::SchemaConfig;
+use slog::Logger;
+use std::collections::BTreeSet;
+use std::ops::Bound;
 use std::str::FromStr;
 
 impl DataStore {
+    // Ensures that the database schema matches "desired_version".
```

Comment:
I think it'd be worth adding more here about what the invariants are around this.  Like:

- Updating the schema version makes the database incompatible with older Nexus instances.
- This is a one-way operation.  It cannot be undone.
- The caller is responsible for ensuring that this will not cause an outage (or something)

Reply:
Will update with these comments